### PR TITLE
Implemented Random Mercenary Company Name Generator

### PR DESCRIPTION
**Pull request by IllianiCBT**
This PR does a couple of things: predominately it implements a random name generator to generate a mercenary company name.

When a player starts a new campaign a random name will be generated in place of the previous 'My Campaign'. Additional company names can be generated via button in Campaign Options. New company names can be generated in active campaigns via the GM Tools dialog. This tech was needed for the upcoming Random Backgrounds Module, as we need to known the origin company name for characters with the 'Mercenary' origin.

Finally, whenever players have a contract against the Mercenary faction mhq will generate a random name for that enemy, instead of using faction name. This tech was needed for a future project currently in pre-development.

There are 6 types of random name: Corporate, Mercenary (Vanity type 1 and 2), Mercenary (Generic), and Pre-Fab (type 1 and 2). If a vanity name is generated mhq will use the name of the current campaign commander, if set, otherwise a random Callsign will be generated.

Examples of the kinds of names generated follows:

- Juggernaut's Centurions
- The Storm Outlaws
- Envision Associates
- Boomtube's Legends
- Raptor Investments
- Barnacle's Wolfpack Gunslingers
- Drillbit's Raven Tactical
- The Starfire Tempests
- Koschei's Midnight Marauders
- The Spectral Legionnaires
- Wallaby's Punchline Platoon
- The Stackpole Survivors

### CODE REVIEW

File: MekHQ/src/mekhq/gui/dialog/GMToolsDialog.java [hunk 1]
```diff
@@ -958,6 +994,100 @@
...
+        return panel;
+    }
+
+    /**
+     * Creates a MMButton object that generates a company name and updates the appropriate JTextArea with the generated name.
+     *
+     * @param txtCompanyNamesGenerated the JTextArea to update with the generated company name
+     * @return a MMButton object that generates a company name and updates the given JTextArea
+     */
+    private MMButton createGenerateNameButton(JTextArea txtCompanyNamesGenerated) {
```
[the remark this answers is not in the document]
Comment (IllianiCBT) [reply MekHQ/src/mekhq/gui/dialog/GMToolsDialog.java:1051]: Resolved

File: MekHQ/resources/mekhq/resources/RandomMercenaryCompanyNameGenerator.properties [hunk 2]
```diff
@@ -0,0 +1,1246 @@
...
+EndWordMercenary.BRIGADE.text=Brigade
+EndWordMercenary.BRIGANDS.text=Brigands
+EndWordMercenary.BULWARKS.text=Bulwarks
+EndWordMercenary.CATAPHRACTS.text=Cataphracts
+EndWordMercenary.CAVALIERS.text=Cavaliers
+EndWordMercenary.CAVALRY.text=Cavalry
+EndWordMercenary.CENTURIONS.text=Centurions
+EndWordMercenary.COMMANDERS.text=Commanders
+EndWordMercenary.COMMANDOS.text=Commandos
+EndWordMercenary.CONQUERORS.text=Conquerors
```
Comment (HammerGS) [inline MekHQ/resources/mekhq/resources/RandomMercenaryCompanyNameGenerator.properties:187]: Cuirassier

File: MekHQ/resources/mekhq/resources/RandomMercenaryCompanyNameGenerator.properties [hunk 3]
```diff
@@ -0,0 +1,1246 @@
...
+EndWordMercenary.BLITZERS.text=Blitzers
+EndWordMercenary.BOUNTY_HUNTERS.text=Bounty Hunters
+EndWordMercenary.BRIGADE.text=Brigade
+EndWordMercenary.BRIGANDS.text=Brigands
+EndWordMercenary.BULWARKS.text=Bulwarks
+EndWordMercenary.CATAPHRACTS.text=Cataphracts
+EndWordMercenary.CAVALIERS.text=Cavaliers
+EndWordMercenary.CAVALRY.text=Cavalry
+EndWordMercenary.CENTURIONS.text=Centurions
+EndWordMercenary.COMMANDERS.text=Commanders
```
Comment (HammerGS) [inline MekHQ/resources/mekhq/resources/RandomMercenaryCompanyNameGenerator.properties:185]: Chevau-léger

**Review by HammerGS — CHANGES_REQUESTED** [review]
Tested and seems to work. Added some of the missing names we see for merc units. 

Also can this generated a numbered unit, i.e 21st Stormbringer Legion

**Comment by IllianiCBT** [comment]
> Tested and seems to work. Added some of the missing names we see for merc units.
> 
> Also can this generated a numbered unit, i.e 21st Stormbringer Legion

It does now! :D

**Comment by IllianiCBT** [comment]
Added numerical name prefixes, allowing us to generate names like "The 22nd Beef Brigade"

**Review by HammerGS — APPROVED** [review]

**Review by HammerGS — APPROVED** [review]
Tested and seems to work

**Pull request merged**